File: Service/Models/Ticket.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace Service.Models;

[Table("Ticket")]
public partial class Ticket
{
    [Key]
    [Column("ID")]
    public long Id { get; set; }

    [Column("TicketNR")]
    public long TicketNr { get; set; }

    [StringLength(50)]
    public string Topic { get; set; } = null!;

    [StringLength(50)]
    public string Name { get; set; } = null!;

    [StringLength(50)]
    public string Firstname { get; set; } = null!;

    [StringLength(50)]
    public string Email { get; set; } = null!;

    public int Priority { get; set; }

    public string Text { get; set; } = null!;

    [Column("Supporter_ID")]
    public int? SupporterId { get; set; }

    [Column(TypeName = "datetime")]
    public DateTime Created { get; set; }

    [InverseProperty("Ticket")]
    public virtual ICollection<Mail> Mail { get; set; } = new List<Mail>();

    [ForeignKey("SupporterId")]
    [InverseProperty("Tickets")]
    public virtual Supporter? Supporter { get; set; }
}

File: frontend/src/Ticket.ts
export class Ticket {
    id: number = 0;
    ticketNr: number = 0;
    topic: string = "";
    name: string = "";
    firstname: string = "";
    email: string = "";
    priority: Priority = Priority.NON_VALUE;
    text: string = "";
    supporterId: number | null = null;
    created: string = "";
}

export class InsertNewTicket{
    email: string = "";
    name: string = "";
    firstname: string = "";
    topic: string = "";
    text: string = "";
}

enum Priority {
        NON_VALUE = 0,
        SEHR_NIEDRIG = 1,
        NIEDRIG = 2,
        MITTEL = 3,
        HOCH = 4,
        SEHR_HOCH = 5
}